### PR TITLE
Add test suite to CI

### PR DESCRIPTION
**Pull request by timcallow**
Attempting to add test suite to CI. Let's see how it goes...

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -65,3 +65,39 @@ jobs:
...
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Upgrade pip
+        run: python3 -m pip install --upgrade pip
+        
+      - name : Install pipenv
+        run : python -m pip install --upgrade pipenv wheel
```
Comment (DanielKotik) [inline .github/workflows/gh-pages.yml:85]: I think `wheel` is not needed here, or  can you explain why its necessary?

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -65,3 +65,39 @@ jobs:
...
+  
+      - id: cache-pipenv
+        uses: actions/cache@v1
+        with:
+          path: ~/.local/share/virtualenvs
+          key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
+     
+      - name: Install dependencies
+        if: steps.cache-pipenv.outputs.cache-hit != 'true'
+        run: pipenv install --deploy --dev
```
Comment (DanielKotik) [inline .github/workflows/gh-pages.yml:95]: I am not sure about the `--deploy` flag. Why is it needed?

**Review by DanielKotik — CHANGES_REQUESTED** [review]
LGTM, but I would suggest to put the test workflow in an extra YAML file in the `.github/workflows/` folder. Maybe call it just `tests.yml`. Then its logically separated from the the workflow for building and deploying the Sphinx documentation deployed at GitHub pages (`gh-pages.yml`). This comes also with more fine control about when test should be triggered (can of course be identical with when docs are triggerd). Start `test.yml` with 

```yaml
name: docs
on:
  pull_request:
    branches:
      - master
      - develop
  push:
    branches:
      - master
      - develop
jobs:
  run-test-suite
```

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (timcallow) [reply .github/workflows/gh-pages.yml]: Tbh I just copied and pasted a guide on running tests as part of a GitHub workflow and kept the commands as similar as possible. Removed `wheel` and everything seems to work fine.

File: .github/workflows/gh-pages.yml (hunk 2 above)
Comment (timcallow) [reply .github/workflows/gh-pages.yml]: See above. Removed it and seems it's not needed.

Comment (DanielKotik) [reply .github/workflows/gh-pages.yml]: Mhh

**Review by DanielKotik — APPROVED** [review]

**Pull request merged**